main.tf

```diff
@@ -7,8 +7,10 @@ locals {
   instance_profile       = local.instance_profile_count == 0 ? var.instance_profile : join("", aws_iam_instance_profile.default.*.name)
   security_group_enabled = module.this.enabled && var.security_group_enabled
   region                 = var.region != "" ? var.region : data.aws_region.default.name
-  root_iops              = var.root_volume_type == "io1" ? var.root_iops : "0"
-  ebs_iops               = var.ebs_volume_type == "io1" ? var.ebs_iops : "0"
+  root_iops              = contains(["io1", "io2", "gp3"], var.root_volume_type) ? var.root_iops : "0"
+  ebs_iops               = contains(["io1", "io2", "gp3"], var.ebs_volume_type) ? var.ebs_iops : "0"
+  root_throughput        = var.root_volume_type == "gp3" ? var.root_throughput : "0"
```

Comment:
instead of using a contains function, you can add validation to root_volume_type and ebs_volume_type to ensure one of these values is present and if it's not you can default to 0

Reply:
see ami type variable validation in eks node groups module for an example

https://github.com/cloudposse/terraform-aws-eks-node-group/blob/37d23d39f4295b841dc0d87775a710b507655c46/variables.tf#L111